### PR TITLE
Fix 6320 Optimize lock when check error storage

### LINKED ISSUE #6320: Disk monitor hold StorageEngine's _store_lock too long
1. `StorageEngine::_delete_tablets_on_unused_root_path` will try to obtain tablet shard write lock in `TabletManager`
```
StorageEngine::_delete_tablets_on_unused_root_path
  TabletManager::drop_tablets_on_error_root_path
    obtain each tablet shard's write lock
```
2. `TabletManager::build_all_report_tablets_info` and other methods will obtain tablet shard read lock frequently.

So, `StorageEngine::_delete_tablets_on_unused_root_path` will hold `_store_lock` for a long time.
This will make it difficult for other threads to get write `_store_lock`, such as `StorageEngine::get_stores_for_create_tablet`

`drop_tablets_on_error_root_path` is a small probability event, `TabletManager::drop_tablets_on_error_root_path` should return when its param `tablet_info_vec` is empty

### PR DESCRIPTION
**Pull request by liutang123**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #6320) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/tablet_manager.cpp [hunk 1]
```diff
@@ -562,13 +562,22 @@ OLAPStatus TabletManager::_drop_tablet_unlocked(TTabletId tablet_id, SchemaHash
 OLAPStatus TabletManager::drop_tablets_on_error_root_path(
         const std::vector<TabletInfo>& tablet_info_vec) {
     OLAPStatus res = OLAP_SUCCESS;
-    for (int32 i = 0; i < _tablets_shards_size; i++) {
+    if (tablet_info_vec.empty()) { // This is a high probability event
+        LOG(INFO) << "tablet_info_vec is empty, return directly.";
```
Comment (morningman) [inline be/src/olap/tablet_manager.cpp:566]: I think this log can be removed?

Comment (liutang123) [reply be/src/olap/tablet_manager.cpp:566]: ok， i will test this pr in our cluster. And then remove this log.

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by liutang123** [comment]
If create tablet thread blocked by report thread, they are mutually exclusive.

This PR may cause the following case:
1. Report thread begin collect tablet info, now report version is v1.
2. Create thread create tablet X, and change the report version to v2.
3. Report thread ends collecting tablet info without X and report all tablet info to FE.
4. FE receive tablet info without X and delete X from meta.

I modified the behavior of tablet report. Before collecting tablet information, mark the report version. After the collection, compare the current report version with the version just marked, if the version has changed, then abandon this report.

File: be/src/agent/task_worker_pool.cpp [hunk 2]
```diff
@@ -1185,9 +1185,15 @@ void TaskWorkerPool::_report_tablet_worker_thread_callback() {
 
         _is_doing_work = true;
         request.tablets.clear();
+        uint64_t report_version = _s_report_version;
```
Comment (morningman) [inline be/src/agent/task_worker_pool.cpp:1188]: using `report_version` in `request`

File: be/src/agent/task_worker_pool.cpp [hunk 3]
```diff
@@ -1185,9 +1185,15 @@ void TaskWorkerPool::_report_tablet_worker_thread_callback() {
...
         OLAPStatus build_all_report_tablets_info_status =
                 StorageEngine::instance()->tablet_manager()->build_all_report_tablets_info(
                         &request.tablets);
+        if (report_version < _s_report_version) {
```
Comment (morningman) [inline be/src/agent/task_worker_pool.cpp:1192]: Add a TODO here to notice that this can only reduce the possibility or report error, but can't avoid it.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**